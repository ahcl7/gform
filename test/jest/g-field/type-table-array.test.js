import { generateComponent } from "./common";

describe('GField with type table array', function () {
  it('should render correctly', () => {
    const component = generateComponent({
      key: 'addressArray3', type: 'tableArray', label: 'Address3',
      fields: [{ key: 'street', type: 'input', flex: 'md-6' }, { key: 'city', type: 'input', flex: 'md-6' },]
    });
    expect(component.html()).toMatchSnapshot();
  });
});
