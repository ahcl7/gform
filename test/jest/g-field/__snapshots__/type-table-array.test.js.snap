// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GField with type table array should render correctly 1`] = `
<div class="col-flex col-xs-12"
     in-array="false"
     no-layout="false"
>
  <table class="g-datatable g-table theme--light gfield-table">
    <thead>
      <tr class="table-header">
        <th>
          Street
        </th>
        <th>
          City
        </th>
        <th>
          X
        </th>
      </tr>
    </thead>
    <tbody>
      <tr class="text-md-center">
        <td class="input-group-sm">
          <input type="text"
                 class="form-control"
          >
        </td>
        <td class="input-group-sm">
          <input type="text"
                 class="form-control"
          >
        </td>
        <td>
          <i class="g-icon material-icons"
             style="font-size: 24px;"
          >
            delete
          </i>
        </td>
      </tr>
      <transition-stub>
      </transition-stub>
      <tr class="text-md-center">
        <td class="input-group-sm">
          <input type="text"
                 class="form-control"
          >
        </td>
        <td class="input-group-sm">
          <input type="text"
                 class="form-control"
          >
        </td>
        <td>
          <i class="g-icon material-icons"
             style="font-size: 24px;"
          >
            delete
          </i>
        </td>
      </tr>
      <transition-stub>
      </transition-stub>
    </tbody>
  </table>
  <div class="g-btn g-btn__flat g-btn__depressed g-btn__outlined g-btn__uppercase text-blue-lighten-2 g-size__small ma-2">
    <span class="g-btn__content">
      Add Address3
    </span>
  </div>
</div>
`;
